HTTPServer.test/ServerTests.cs

```diff
@@ -25,14 +25,14 @@ public class ServerUnitTests
         [TestMethod]
         public void ServerCanStart()
         {
-            Assert.True(server.Start(8080, new NetworkSocket()) != null && server.Running);
+            Assert.True(server.Start(8080, new NetworkSocket(), "") != null && server.Running);
```

Comment:
It's strange to me that `server.start()` would return the server or `null`. Returning `null` is still you indicating some sort of error condition. I don't think starting a server which is already started should be an error.